packages/framework/src/client.ts

```diff
@@ -628,7 +629,11 @@ export class Client {
           symbol: EMOJI.ERROR,
           text: `Failed to execute stepId: \`${step.stepId}\``,
         });
-        throw error;
+        if (isFrameworkError(error)) {
+          throw error;
+        } else {
+          throw new StepExecutionFailedError(step.stepId, event.action, error);
+        }
```

Comment:
Let the original error bubble up. We choose not to encapsulate the `isFrameworkError` check in the `StepExecutionFailedError` constructor logic because doing so necessitates putting it behind the `new` keyword, which breaks the stack-trace of the original error.